lighthouse-core/audits/unused-css-rules.js

```diff
@@ -79,6 +89,42 @@ class UnusedCSSRules extends Audit {
   }
 
   /**
+   * Trims stylesheet content down to the first rule-set definition.
+   * @param {string} content
+   * @return {string}
+   */
+  static determineContentPreview(content) {
+    let preview = content
+        .slice(0, PREVIEW_LENGTH * 5)
+        .replace(/( {2,}|\t)+/g, '  ') // remove leading indentation if present
+        .replace(/\n\s+}/g, '\n}') // completely remove indentation of closing braces
+        .trim(); // trim the leading whitespace
+
+    if (preview.length > PREVIEW_LENGTH) {
+      const firstRuleStart = preview.indexOf('{');
+      const firstRuleEnd = preview.indexOf('}');
+
+      if (firstRuleStart === -1 || firstRuleEnd === -1
```

Comment:
Yeah I tried to use these but because everything is expressed in terms of lines and columns translating to absolute length ended up being even more complicated than this (and didn't really handle leading comments such as those used in every lighthouse file :) ) It also focuses a lot more on rules than rule-sets which made for tricky and generally confusing logic to show the first rule-set with its selector.